extensions-core/multi-stage-query/src/main/java/org/apache/druid/msq/util/MultiStageQueryContext.java

```diff
@@ -377,9 +377,17 @@ public static boolean removeNullBytes(final QueryContext queryContext)
     return queryContext.getBoolean(CTX_REMOVE_NULL_BYTES, DEFAULT_REMOVE_NULL_BYTES);
   }
 
+  public static boolean isDartQuery(final QueryContext queryContext)
+  {
+    return queryContext.get(QueryContexts.CTX_DART_QUERY_ID) != null;
+  }
 
   public static MSQSelectDestination getSelectDestination(final QueryContext queryContext)
   {
+    if (isDartQuery(queryContext)) {
+      // Dart does not support any other destination.
+      return MSQSelectDestination.TASKREPORT;
```

Comment:
Lets add a log.warn here that we donot support any other destination than task report in DART.